react/features/base/conference/actions.js

```diff
@@ -146,6 +154,25 @@ function _addConferenceListeners(conference, dispatch, state) {
             }
         });
 
+    conference.on(
+        JitsiConferenceEvents.AUDIO_UNMUTE_PERMISSIONS_CHANGED,
+        disableAudioMuteChange => {
+            const muted = isAudioMuted(state);
+
+            if (!disableAudioMuteChange || (disableAudioMuteChange && muted)) {
+                APP.store.dispatch(setAudioAvailable(!disableAudioMuteChange));
```

Comment:
is `setAudioAvailable` a new method? or is it the one that was used to tell if there are audio devices available?

I think there should be dedicated state for "permission to unmute audio". I think mixing both together is dangerous, in a way that it won't be possible to distinguish between unmute permission and the audio device availability (to show different UI for each case). 

The `if` condition here will become much simpler where in fact will be just a setter. Then in the UI you can do a combined selector for "is audio device available" and "has permission to unmute".

Reply:
The UI is going to be the same for both the cases so went ahead and used the existing one but I agree its better to have a dedicated state for this. I will update the PR.